genesyscloud/data_source_genesyscloud_routing_skill.go

```diff
@@ -28,24 +32,67 @@ func dataSourceRoutingSkill() *schema.Resource {
 	}
 }
 
-func dataSourceRoutingSkillRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	const pageSize = 100
-	var pageCount int
+var (
+	dataSourceRoutingSkillCache *rc.DataSourceCache
+)
 
+func dataSourceRoutingSkillRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	sdkConfig := m.(*provider.ProviderMeta).ClientConfig
-	routingAPI := platformclientv2.NewRoutingApiWithConfig(sdkConfig)
-	name := d.Get("name").(string)
 
-	skills, resp, getErr := routingAPI.GetRoutingSkills(pageSize, 1, name, nil)
-	if getErr != nil {
-		return util.BuildAPIDiagnosticError("genesyscloud_routing_skill", fmt.Sprintf("Error requesting skills error: %s", getErr), resp)
+	key := d.Get("name").(string)
+
+	if dataSourceRoutingSkillCache == nil {
+		dataSourceRoutingSkillCache = rc.NewDataSourceCache(sdkConfig, hydrateRoutingSkillCacheFn, getSkillByNameFn)
 	}
-	pageCount = *skills.PageCount
+
+	queueId, err := rc.RetrieveId(dataSourceRoutingSkillCache, "genesyscloud_routing_skill", key, ctx)
+
+	if err != nil {
+		return err
+	}
+
+	d.SetId(queueId)
+	return nil
+}
+
+func hydrateRoutingSkillCacheFn(c *rc.DataSourceCache) error {
+	log.Printf("hydrating cache for data source genesyscloud_routing_skill")
+
+	routingApi := platformclientv2.NewRoutingApiWithConfig(c.ClientConfig)
+
+	for pageNum := 1; ; pageNum++ {
```

Comment:
Should we be sticking to the new way of doing this? Where we do one api call first to get the number of pages and then loop up to that

Reply:
I thought to do it that way, but really find little advantage more than an additional call.  what do you think ?